src/gridfs/upload.ts

```diff
@@ -131,14 +140,11 @@ export class GridFSBucketWriteStream extends Writable {
       aborted: false
     };
 
-    if (!this.bucket.s.calledOpenUploadStream) {
-      this.bucket.s.calledOpenUploadStream = true;
-
-      checkIndexes(this).then(() => {
-        this.bucket.s.checkedIndexes = true;
-        this.bucket.emit('index');
-      }, squashError);
-    }
+    if (options.timeoutMS != null)
+      this.timeoutContext = new CSOTTimeoutContext({
```

Comment:
Any reason we're not using `TimeoutContext.create()`?

Reply:
The only time we'll have a timeout context on a GridFSStream is when CSOT is enabled, so we don't need the broader abstraction here.